examples/example_dotnet/WinSparkleDemoForm.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Reflection;
+using System.Threading;
+using System.Windows.Forms;
+using WinSparkleDotNet;
+
+namespace example_dotnet
+{
+    public partial class WinSparkleDemoForm : Form
+    {
+        private SynchronizationContext _context;
+        private static string _appCastURL = @"http://winsparkle.org/example/appcast.xml";
+
+        public WinSparkleDemoForm()
+        {
+            _context = SynchronizationContext.Current;
+            InitializeComponent();
+            
+            var assembly = Assembly.GetExecutingAssembly();
+            var company = (AssemblyCompanyAttribute)assembly.GetCustomAttributes(typeof(AssemblyCompanyAttribute), false)[0];
+            var name = assembly.GetName().Name;
+            var version = assembly.GetName().Version; //Sparkle can parse basic version scheme like X.X.X.X
+
+            WinSparkle.SetAppcastUrl(_appCastURL);
+            WinSparkle.SetAppDetails(company.Company, name, version.ToString());
```

Comment:
> Just before you said that 4 lines show as "pretty complicated”

Please. I was, clearly I would think, referring to *code using WinSparkle* and necessary to accomplish the most simple use there. That doesn’t imply in any way that the *library implementation* needs to be simple.

To avoid any ambiguity, let me restate the obvious: it’s good to do more work in the library than in user code. It’s better to have more complicated code in WinSparkle *once* than to do the same 4-lines little dance a thousand times in every use of the assembly.